lib/widgets/board_icon.dart

```diff
@@ -8,10 +9,13 @@ import '../model/game.dart';
 /// It holds board value named [board].
 class BoardIconView extends StatelessWidget {
   const BoardIconView({super.key, required this.board});
+
   final Board board;
 
   @override
   Widget build(BuildContext context) {
+    final int boardLength = board.length;
+    boardSize = boardLength;
```

Comment:
Do not store boardSize as global variable. Pass it to constructor of RulesChangingButtons instead.